pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/reader/JsonIndexReader.java

```diff
@@ -35,16 +35,28 @@ public interface JsonIndexReader extends IndexReader {
   MutableRoaringBitmap getMatchingDocIds(String filterString);
 
   /**
-   * For an array of docIds and context specific to a JSON key, returns the corresponding values for each docId. The
-   * context should be created from the getMatchingDocsMap method.
-   *
-   * @return String[] where String[i] is the value for docIds[i]
+   * For an array of docIds and context specific to a JSON key, returns the corresponding mv array for each docId.
+   * @param docIds array of docIds
+   * @param length length of the array
+   * @param matchingValueToFlattenedDocs Map from each unique value for the jsonPathKey value to the flattened docId
+   *                                     posting list
+   * @return String[][] where String[i] is the mv array for docIds[i]
    */
-  String[] getValuesForKeyAndDocs(int[] docIds, Map<String, RoaringBitmap> context);
+  String[][] getValuesForMv(int[] docIds, int length, Map<String, RoaringBitmap> matchingValueToFlattenedDocs);
```

Comment:
```suggestion
  String[][] getValuesMV(int[] docIds, int length, Map<String, RoaringBitmap> matchingValueToFlattenedDocs);
```

Reply:
(nit) We usually put API for SV in front of MV

Reply:
Ack